### ISSUE TITLE
simpler preview mode?

### ISSUE THREAD

**Issue by colemickens** [kamiyaa/joshuto#146]
First, a small note:  `use_preview_script` isn't listed in the example config, but seems like it's maybe needed to get previews to work.

I think it would be nice to have a simpler preview mode. For example `bat` already knows how to preview many types of files. It would be nice if `joshuto` could use `bat` as a lib or have a simpler mode where it can just use a single executable to naive handle any file type.

**Comment by kamiyaa** [comment]
I did a bit of research on this in the past when scoping out file preivews. And from what I've gathered, `bat` doesn't provide any libraries other rust programs can plug into.

So the only viable solution is to parse bat's output and convert that (ansi codes) into tui style structs.

**Comment by alexzanderr** [comment]
Hello.

Is the preview.sh` is run async?

And is the preview of a file cached, somehow ? Cuz I saw a white bar for every file which was previewed, but the second time when previewing, the white bar was no longer there ?

**Comment by DLFW** [comment]
Hi!
Yes and yes.
That's why the preview pane disappears for a very short time when the cursor moves to a not-yet-visited file and is then shown again once the preview text has been provided by `preview.sh`. (Assuming that the default configuration `collapse_preview = true` is used.)

Not really sure though what “white bar” you mean...

**Comment by alexzanderr** [comment]
yeah. `collapse_preview = true`. thanks for response. i understand now
